tasmota/tasmota_xsns_sensor/xsns_111_ens16x.ino

```diff
@@ -0,0 +1,220 @@
+/*
+  xsns_98_ENS16x.ino - ENS16x gas and air quality sensor support for Tasmota
+
+  Copyright (C) 2021  Christoph Friese and Theo Arends
+
+  This program is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifdef USE_I2C
+#ifdef USE_ENS16X
+
+/*********************************************************************************************\
+ * ENS16x - Gas (TVOC - Total Volatile Organic Compounds) and Air Quality (CO2)
+ *
+ * Source: ScioSense
+ *
+ * I2C Address: 0x52 
+\*********************************************************************************************/
+
+#define XSNS_111             111
+#define XI2C_84             84  // See I2CDEVICES.md
+
+#define EVERYNSECONDS 5
+
+#include "ScioSense_ENS16x.h"
+
+ScioSense_ENS16x      ens16xa(ENS16x_I2CADDR_0);
+struct  {
+  uint16_t  TVOC;
+  uint16_t  eCO2;
+  uint16_t  AQIS;
+
+  uint8_t ready = 0;
+  uint8_t type = 0;;
+  uint8_t tcnt = 0;
+  uint8_t ecnt = 0;
+} ENS16xdataA;
+
+ScioSense_ENS16x      ens16xb(ENS16x_I2CADDR_1);
+struct  {
+  uint16_t  TVOC;
+  uint16_t  eCO2;
+  uint16_t  AQIS;
+
+  uint8_t ready = 0;
+  uint8_t type = 0;;
+  uint8_t tcnt = 0;
+  uint8_t ecnt = 0;
+} ENS16xdataB;
```

Comment:
Use dynamically allocated memory
This is using memory in any built your driver is included even if no-one is using it

Reply:
Thanks for proposal. I'll try to implement later.